src/mesh/scalar_field.rs

```diff
@@ -0,0 +1,1899 @@
+                    let ca_distance_sq = nalgebra::distance_squared(point_c, point_a);
+                    let longest_edge_len = ab_distance_sq
+                        .max(bc_distance_sq.max(ca_distance_sq))
+                        .sqrt();
+                    // Number of face divisions (points) in each direction.
+                    let divisions = (longest_edge_len / smallest_voxel_dimension).ceil() as usize;
+                    let divisions_f32 = divisions as f32;
+
+                    for ui in 0..=divisions {
+                        for wi in 0..=divisions {
+                            let u_normalized = ui as f32 / divisions_f32;
+                            let w_normalized = wi as f32 / divisions_f32;
+                            let v_normalized = 1.0 - u_normalized - w_normalized;
+                            if v_normalized >= 0.0 {
+                                let barycentric =
+                                    Point3::new(u_normalized, v_normalized, w_normalized);
+                                // Compute point position in model space
+                                let cartesian = geometry::barycentric_to_cartesian(
+                                    &barycentric,
+                                    &point_a,
+                                    &point_b,
+                                    &point_c,
+                                );
+                                // and set a voxel containing the point to be on
+                                let absolute_coordinate = cartesian_to_absolute_voxel_coordinate(
+                                    &cartesian,
+                                    voxel_dimensions,
+                                );
+                                scalar_field.set_value_at_absolute_voxel_coordinate(
+                                    &absolute_coordinate,
+                                    Some(value_on_mesh_surface),
+                                );
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        scalar_field
+    }
+
+    /// Creates a new scalar field with arbitrary voxel dimensions from another
+    /// scalar field.
+    ///
+    /// # Panics
+    /// Panics if any of the voxel dimensions is below or equal to zero.
+    pub fn from_scalar_field(
+        source_scalar_field: &ScalarField<T>,
+        volume_value_interval: Interval<T>,
+        voxel_dimensions: &Vector3<f32>,
+    ) -> Option<Self> {
+        assert!(
+            voxel_dimensions.x > 0.0 && voxel_dimensions.y > 0.0 && voxel_dimensions.z > 0.0,
+            "One mor more voxel dimensions is below or equal to zero"
+        );
+
+        // Make a bounding box of the source bounding box's mesh volume. This
+        // will be the volume to be scanned for any voxels.
+        source_scalar_field
+            .mesh_volume_bounding_box_cartesian(volume_value_interval)
```

Comment:
Because the new scalar field should be able to hit all the materialized voxels, which means it needs to scan the volume occupied by the meshed voxels.